### PR TITLE
Add `is_partitioned` column

### LINKED ISSUE #871: [FEATURE]: Add "is_partitioned" field to HMS tables crawler output
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

Special consideration needs to be taken when migrating HMS partitioned tables to UC external table. Currently there is no way to distinguish between HMS partitioned and non-partitioned, based on the output of the assessment.

### Proposed Solution

I see two files tables.py and tables.scala, not sure if both need to be modified or just the scala version.
For the Python version, DESCRIBE EXTENDED outputs partition information if applicable, and could be used to derive is_partitioned.
For the Scala version, the attached piece of code seems to do the trick. I decided not to submit a PR because I am weak in Scala and also this involves a separate function call (getTableMetadata - see the attached notebook) . The notebook also includes two other alternative approaches that did not work.
[test_partitioned.txt](https://github.com/databrickslabs/ucx/files/14131431/test_partitioned.txt)

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dleiva04**
#  Changes

### Linked issues

Resolves #871 

Related PR:
- https://github.com/databrickslabs/ucx/pull/985

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [X] modified existing table: `ucx.tables`

### Tests

- [X] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

image
image

### CODE REVIEW

File: src/databricks/labs/ucx/upgrades/v0.19.0_add_is_partitioned_column.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# pylint: disable=invalid-name
```
Comment (nfx) [inline src/databricks/labs/ucx/upgrades/v0.19.0_add_is_partitioned_column.py:1]: it has to be v0.20.0 now :)

**Review by nfx — CHANGES_REQUESTED** [review]

**Comment by nfx** [comment]
okay, merging

File: src/databricks/labs/ucx/upgrades/v0.19.0_add_is_partitioned_column.py (hunk 1 above)
Comment (dleiva04) [reply src/databricks/labs/ucx/upgrades/v0.19.0_add_is_partitioned_column.py:1]: Oh ok

**Review by nfx — APPROVED** [review]
please do a follow-up rename of the upgrade file

**Pull request merged**